### DISCUSSION TITLE
Problem with "loading" in "when"

### DISCUSSION THREAD

**Discussion by jamolina2021** [rrousselGit/riverpod#1177] [Q&A]
Hi,
The loading doesn't seem to run when an error occurs.
For example, in the following code snippet:
```
return ref.watch(dashboardInfo).when(
        loading: () => Container(
              color: Colors.white,
              child: const Center(child: CircularProgressIndicator()),
            ),
        error: (err, stack) {
          var msg = '';
          if (err is DioError) {
            msg = err.error;
          } else {
            msg = err.toString();
          }
          return Center(child: Text('Error: $msg'));
        },
        data: (dashboardInfo) {
```
Dio does a GET, but the Rest API server is stopped, so an error occurs: **XMLHttpRequest error**
The error parameter, after a certain time, is executed correctly, but the **CircularProgressIndicator is not displayed** in the time interval before the error parameter is executed. 
Is this a bug? 

Thanks for your help.
Regards,
Jose

**Comment by zaprogrammer** [comment]
Hi have the same issue, I have a similar code inside a button press, when I first press the button, it triggers loading(), network request and response and done, but data() is not triggered, I have to press the button 1 more time to fetch the data, which is fetched from the state (not network).

```dart
TextButton(
  onPressed: () {
    ref.read(userProvider).map(
        data: (data) => {
              print(
                  'My data: ${data.username}'),
              _isLoading = false
            },
        error: (e) => print('Error => $e'),
        loading: () {
          _isLoading = true;
        });
}),

if (_isLoading) const CircularProgressIndicator(),
```

And this is my future provider:
```dart
final userProvider = FutureProvider<User>((ref) async {
  return login(username, password);
});
```

**Comment by rrousselGit** [comment]
I'll need more information. 

Could you share the version of Riverpod you're using and a minimal reproducible example?
